chia/rpc/wallet_rpc_api.py

```diff
@@ -1066,13 +1067,62 @@ async def cancel_offer(self, request: Dict) -> EndpointResult:
         secure = request["secure"]
         trade_id = bytes32.from_hexstr(request["trade_id"])
         fee: uint64 = uint64(request.get("fee", 0))
-
+        txs = None
         async with self.service.wallet_state_manager.lock:
             if secure:
-                await wsm.trade_manager.cancel_pending_offer_safely(bytes32(trade_id), fee=fee)
+              txs = await wsm.trade_manager.cancel_pending_offer_safely(bytes32(trade_id), fee=fee)
             else:
                 await wsm.trade_manager.cancel_pending_offer(bytes32(trade_id))
-        return {}
+        return {"success": True, "TXs": txs}
+
+    async def cancel_offers(self, request: Dict) -> EndpointResult:
+        secure = request["secure"]
+        fee: uint64 = uint64(request.get("fee", 0))
+        batch_size = request.get("batch_size", 5)
+        cancel_all = request.get("cancel_all", False)
+        if cancel_all:
+            asset_id = None
+        else:
+            asset_id = request.get("asset_id", "xch")
+
+        start: int = 0
+        end: int = start + batch_size
+        trade_mgr = self.service.wallet_state_manager.trade_manager
+        log.info(f"Start cancelling offers for asset_id {asset_id} ...")
+        # Traverse offers page by page
+
+        while True:
+            records: List[TradeRecord] = []
+            trades = await trade_mgr.trade_store.get_trades_between(
+                start,
+                end,
+                reverse=True,
+                exclude_my_offers=False,
+                exclude_taken_offers=True,
+                include_completed=False,
+            )
+            for trade in trades:
+                if cancel_all:
+                    records.append(trade)
+                    continue
+                if trade.offer and trade.offer != b'':
+                    summary = Offer.from_bytes(trade.offer).summary()
+                    if asset_id == list(summary[0].keys())[0]:
+                        records.append(trade)
+                        continue
+                    if asset_id == list(summary[1].keys())[0]:
+                        records.append(trade)
+                        continue
```

Comment:
The `summary` method is really just a display method, you shouldn't use it for business logic.  `if bytes32.from_hexstr(asset_id) in offer.driver_dict` is a better check (The asset_id for XCH in that case is None).